Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/UpgradeData.xml

```diff
@@ -12792,4 +12822,35 @@
         <EditorCategories value="Race:Zerg,UpgradeType:Talents"/>
     </CUpgrade>
     <CUpgrade id="AP_VoidImmortalWeaponSplash"/>
+    <CUpgrade id="AP_TyrannozorTyrantsProtection">
+        <AffectedUnitArray value="AP_Tyrannozor"/>
+        <AffectedUnitArray value="AP_TyrannozorBurrowed"/>
+        <Icon value="Assets\Textures\btn-ability-zerg-dehaka-tyrannozor-tyrantprotection.dds"/>
+    </CUpgrade>
+    <CUpgrade id="AP_TyrannozorBarrageOfSpikes">
+        <AffectedUnitArray value="AP_Tyrannozor"/>
+        <Icon value="Assets\Textures\btn-ability-zerg-dehaka-tyrannozor-barrageofspikes.dds"/>
+    </CUpgrade>
+    <CUpgrade id="AP_ImpalingStrike">
+        <AffectedUnitArray value="AP_Ultralisk"/>
+        <AffectedUnitArray value="AP_Tyrannozor"/>
+        <Icon value="Assets\Textures\btn-ability-zerg-dehaka-ultralisk-impalingstrike.dds"/>
+    </CUpgrade>
+    <CUpgrade id="AP_HealingAdaptation">
+        <AffectedUnitArray value="AP_Ultralisk"/>
+        <AffectedUnitArray value="AP_UltraliskBurrowed"/>
+        <AffectedUnitArray value="AP_Tyrannozor"/>
+        <AffectedUnitArray value="AP_TyrannozorBurrowed"/>
+        <EffectArray Reference="Unit,AP_Ultralisk,LifeRegenRate" Value="10.000000"/>
+        <EffectArray Reference="Unit,AP_UltraliskBurrowed,LifeRegenRate" Value="10.000000"/>
+        <EffectArray Reference="Unit,AP_UltraliskResourceEfficiency,LifeRegenRate" Value="10.000000"/>
+        <EffectArray Reference="Unit,AP_Ultralisk,LifeRegenDelay" Value="10.000000"/>
+        <EffectArray Reference="Unit,AP_UltraliskBurrowed,LifeRegenDelay" Value="10.000000"/>
+        <EffectArray Reference="Unit,AP_UltraliskResourceEfficiency,LifeRegenDelay" Value="10.000000"/>
+        <EffectArray Reference="Unit,AP_Tyrannozor,LifeRegenRate" Value="10.000000"/>
+        <EffectArray Reference="Unit,AP_TyrannozorBurrowed,LifeRegenRate" Value="10.000000"/>
+        <EffectArray Reference="Unit,AP_Tyrannozor,LifeRegenDelay" Value="10.000000"/>
+        <EffectArray Reference="Unit,AP_TyrannozorBurrowed,LifeRegenDelay" Value="10.000000"/>

```

Comment:
Bug: Stops regular life regen (that works even in combat). Look into Regen Biosteel code for Infested Tanks/Banshees that allows basic regen (and increased regen while out of combat)

Reply:
Fixed.  Though I'll note that I just copied vanilla on this originally, which is apparently kinda dumb.